source/include/FreeRTOS_IPv4.h

```diff
@@ -43,9 +43,17 @@
 
 /* Forward declarations. */
 struct xNETWORK_BUFFER;
-enum eFrameProcessingResult;
 struct xIP_PACKET;
 
+#ifdef __GNUC__
```

Comment:
Why is this needed, i see that by adding -Wpedantic to the GCC compilation flags only FreeRTOS_Sockets.c issue is seen.

Reply:
It gives an error on GCC 13.2

```
/opt/arm/gcc-13.2/bin/arm-none-eabi-gcc 

-mcpu=cortex-m7 -mfpu=fpv5-d16 -mfloat-abi=hard -mthumb -mlong-calls -mlittle-endian -Wall -Wextra -Werror -Wpedantic -Wno-psabi -fstack-usage -ffunction-sections -fdata-sections -fno-omit-frame-pointer -fvar-tracking -fvar-tracking-assignments -O0 -DDEBUG  -ggdb3 -gdwarf-5 -std=gnu2x -fdiagnostics-color=always -Wno-array-bounds -Wno-strict-aliasing -Wno-unused-but-set-variable

...
In file included from .../FreeRTOS_IP.h:496,
                 from .../FreeRTOS_DNS_Networking.h:30,
                 from .../FreeRTOS_DNS_Networking.c:36:
FreeRTOS_IPv4.h:50:10: error: empty declaration of 'enum' type does not redeclare tag [-Werror=pedantic]
   50 |     enum eFrameProcessingResult;
``